x/ecocredit/server/marketplace/sell_test.go

```diff
@@ -0,0 +1,150 @@
+package marketplace
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/golang/mock/gomock"
+	marketApi "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	ecocreditv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	v1 "github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gotest.tools/v3/assert"
+	"testing"
+	"time"
+)
+
+func TestSell_Valid(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	batchDenom := "C01-20200101-20200201-001"
+	start, end := timestamppb.Now(), timestamppb.Now()
+	ask := sdk.NewInt64Coin("ufoo", 10)
+	creditType := ecocredit.CreditType{
+		Name:         "carbon",
+		Abbreviation: "C",
+		Unit:         "tonnes",
+		Precision:    6,
+	}
+	testSellSetup(t, s, batchDenom, ask.Denom, ask.Denom[1:], "C01", start, end, creditType)
```

Comment:
Ok to leave as is but you could also return credit type and include declaration in setup rather than having to redeclare the credit type in each test.